### PR TITLE
adding additional queue logging

### PR DESCRIPTION
**Pull request by brettsam**
We're getting a lot of CRIs that require us to figure out why queue messages are being missed or why they're taking longer than expected. We don't really have any insight into this today. I'm adding two new logs that will help us be able to diagnose these problems.

@FinVamp1 FYI... this may be useful for you.

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Listeners/QueueListener.Logger.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        private static class Logger
+        {
+            private static readonly Action<ILogger, string, string, string, int, long, Exception> _getMessages =
+                LoggerMessage.Define<string, string, string, int, long>(LogLevel.Debug, new EventId(1, nameof(GetMessages)),
+                    "Poll for function '{functionName}' on queue '{queueName}' with ClientRequestId '{clientRequestId}' found {messageCount} messages in {pollLatency} ms.");
+
+            public static void GetMessages(ILogger logger, string functionName, string queueName, string clientRequestId, int messageCount, long pollLatency) =>
+                _getMessages(logger, functionName, queueName, clientRequestId, messageCount, pollLatency, null);
+
+            private static readonly Action<ILogger, string, double, string, Exception> _backoffDelay =
```
Comment (alrod) [inline src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Listeners/QueueListener.Logger.cs]: Can you please move it up so private fields will be in the beginning.

**Review by alrod — APPROVED** [review]
Approved with a minor comment

File: src/Microsoft.Azure.WebJobs.Extensions.Storage/StorageLoadBalancerQueue.cs [hunk 2]
```diff
@@ -111,6 +111,7 @@ private CloudQueue Convert(string queueMoniker)
                 sharedWatcher: _sharedWatcher,
                 queueOptions: _queueOptions,
                 queueProcessorFactory: _queueProcessorFactory,
+                functionDescriptor: new FunctionDescriptor(),
```
Comment (brettsam) [inline src/Microsoft.Azure.WebJobs.Extensions.Storage/StorageLoadBalancerQueue.cs]: I'm pretty sure that this isn't used anywhere anymore -- but even if so, I believe it was meant to be a "backplane" for queues across our system, so having (null) in the function name would make sense. I didn't want to branch the logging logic based on this, though -- so leaving as-is for now. If we end up seeing this as a problem, we can address it with a follow-up.

**Comment by FinVamp1** [comment]
Thanks Brett, adding this to https://github.com/Azure/azure-webjobs-sdk/issues/2269 as that was my ask there.

**Pull request merged**